### ISSUE TITLE
[1.17.1] Bars reset

### ISSUE THREAD

**Issue by Silence15** [dannydjdk/Nutritional-Balance#9]
I have seen an issue where entering the nether resets the bars to 50. Then once you return, it bars reset again. I am using nutritionalbalance-1.17.1-2.0.0a.jar on forge-1.17.1-37.0.71.

**Comment by dannydjdk** [comment]
Thanks for the report! This is actually a known issue that's mentioned in the change log, but having a formal issue open for it is a good idea. The problem also exists on death.

This appears to be a forge bug with player capabilities. I opened up a discussion about it here: https://forums.minecraftforge.net/topic/103197-1171-player-capabilities-are-invalidated-on-death-before-they-can-be-copied/?tab=comments#comment-462937

I am considering a workaround involving a different method of storing this data, so I don't have to rely on forge capabilities. 

I have found that if you go to the nether, then return, then leave the game and rejoin, the data will usually be restored. It's an annoying, clunky workaround, and it doesn't work on death or return from The End, so hopefully, I'll have a fix ready for this soon if forge doesn't.
